compiler/llvm/llvmAggregateGlobalOps.cpp

```diff
@@ -831,21 +827,14 @@ Instruction *AggregateGlobalOpsOpt::tryAggregating(Instruction *StartInst, Value
         Value* i8Src = irBuilder.CreateInBoundsGEP(int8Ty,
                                                    alloc,
                                                    offsets);
-#if HAVE_LLVM_VER >= 150
-        Type* SrcTy = oldLoad->getType();
-#else
-        Type* origSrcTy = oldLoad->getPointerOperand()->getType();
-        Type* SrcTy = origSrcTy->getPointerElementType()->getPointerTo(0);
-#endif
-        Value* Src = irBuilder.CreatePointerCast(i8Src, SrcTy);
 
-#if HAVE_LLVM_VER >= 150
-        LoadInst* newLoad = irBuilder.CreateLoad(SrcTy, Src);
-#elif HAVE_LLVM_VER >= 130
-        LoadInst* newLoad = irBuilder.CreateLoad(Src->getType()->getPointerElementType(), Src);
-#else
-        LoadInst* newLoad = irBuilder.CreateLoad(Src);
-#endif
+        Type* LoadType = oldLoad->getType();
+        Type* PtrType = LoadType->getPointerTo(0);
+
+        Value* Src = irBuilder.CreatePointerCast(i8Src, PtrType);
+
+        LoadInst* newLoad = irBuilder.CreateLoad(LoadType, Src);
```

Comment:
Previously for LLVM <= 13 we were giving `CreateLoad` just one argument, is this invocation also valid in LLVM 11-13?

Reply:
Yes, LLVM 11 has this form of CreateLoad accepting a type and a value.